### PR TITLE
temporary fix of VirusTotal offset 1 error in reputation

### PR DESCRIPTION
**Pull request by thefrieddan1**
scripts

## Contributing to Cortex XSOAR Content
Make sure to register your contribution by filling the [contribution registration form](https://forms.gle/XDfxU4E61ZwEESSMA)

**The Pull Request will be reviewed only after the contribution registration form is filled.**

## Status
- [x] In Progress
- [ ] Ready
- [ ] In Hold - (Reason for hold)

## Related Issues
fixes: https://jira-hq.paloaltonetworks.local/browse/XSUP-17626

## Description
Ignore error msg: "offset: 1".
The error returns from VT integration called from reputation scripts ( File, URL, Domain).
A Temporary fix.

## Screenshots
Paste here any images that will help the reviewer

## Does it break backward compatibility?
   - [ ] Yes
       - Further details:
   - [x] No

## Must have
- [ ] Tests
- [ ] Documentation

### CODE REVIEW

File: Packs/CommonScripts/ReleaseNotes/1_9_8.md [hunk 1]
```diff
@@ -0,0 +1,10 @@
+#### Scripts
+##### DomainReputation
+- Fixed an issue where calling VirusTotal integration would return error msg offset 1.
```
Comment (ShahafBenYakir) [inline Packs/CommonScripts/ReleaseNotes/1_9_8.md:3]:
```suggestion
- Fixed an issue where using the script together with VirusTotal integration would return an error.
```

File: Packs/CommonScripts/Scripts/DomainReputation/DomainReputation.py [hunk 2]
```diff
@@ -6,12 +6,19 @@ def domain_reputation():
...
     for item in results:
-        if isError(item):
+        if isError(item) and is_valid_error(item):  # call to is_valid_error is a temporary fix to ignore offset 1 error
             item['Contents'] = item['Brand'] + ' returned an error.\n' + str(item['Contents'])
 
     demisto.results(results)
 
 
+# remove this method once XSUP-18208 is fixed in ExecutionMetrics / Server
+def is_valid_error(item) -> bool:
```
Comment (ShahafBenYakir) [inline Packs/CommonScripts/Scripts/DomainReputation/DomainReputation.py:16]: let's add small doc string

File: Packs/CommonScripts/Scripts/DomainReputation/DomainReputation.py [hunk 3]
```diff
@@ -6,12 +6,19 @@ def domain_reputation():
...
+    if item['Brand'] == 'VirusTotal (API v3)' and item['Contents'] == "'Offset': 1":
```
Comment (ShahafBenYakir) [inline Packs/CommonScripts/Scripts/DomainReputation/DomainReputation.py:17]: Why can't we have this fix for all integrations by removing `if item['Brand'] == 'VirusTotal (API v3)'` and keeping the second part of: `item['Contents'] == "'Offset': 1"`

File: Packs/CommonScripts/Scripts/DomainReputation/DomainReputation_test.py [hunk 4]
```diff
@@ -19,3 +19,24 @@ def test_domain_reputation(mocker):
     domain_reputation()
     assert execute_mock.call_count == 1
     assert 'returned an error' in results_mock.call_args[0][0][0]['Contents']
+    from DomainReputation import domain_reputation
+
+
+def test_domain_reputation_ignore_offset_error(mocker):
+    """
+    Given:
+        - The script args.
```
Comment (ShahafBenYakir) [inline Packs/CommonScripts/Scripts/DomainReputation/DomainReputation_test.py:28]: Let's mention the script args

File: Packs/CommonScripts/Scripts/DomainReputation/DomainReputation_test.py [hunk 5]
```diff
@@ -19,3 +19,24 @@ def test_domain_reputation(mocker):
...
+
+    When:
+        - Running ip_reputation function.
```
Comment (ShahafBenYakir) [inline Packs/CommonScripts/Scripts/DomainReputation/DomainReputation_test.py:31]: You are missing the important part here
```suggestion
        - Running ip_reputation function using VT integration and an error entry (type 4) of "offset 1" is returned.
```

File: Packs/CommonScripts/Scripts/DomainReputation/DomainReputation_test.py [hunk 6]
```diff
@@ -19,3 +19,24 @@ def test_domain_reputation(mocker):
...
+    Then:
+        - Ignores offset 1 error.
```
Comment (ShahafBenYakir) [inline Packs/CommonScripts/Scripts/DomainReputation/DomainReputation_test.py:33]:
```suggestion
        - Ensure the script will ignore the offset 1 error.
```

**Review by ShahafBenYakir — COMMENTED** [review]
Nice. 
Gave you comments on one of the scripts and tests but the comments are relevant to all 3.

File: Packs/CommonScripts/Scripts/DomainReputation/DomainReputation_test.py (hunk 6 above)
Comment (thefrieddan1) [reply Packs/CommonScripts/Scripts/DomainReputation/DomainReputation_test.py:33]: done.

File: Packs/CommonScripts/Scripts/DomainReputation/DomainReputation_test.py (hunk 5 above)
Comment (thefrieddan1) [reply Packs/CommonScripts/Scripts/DomainReputation/DomainReputation_test.py:31]: done

File: Packs/CommonScripts/Scripts/DomainReputation/DomainReputation_test.py (hunk 4 above)
Comment (thefrieddan1) [reply Packs/CommonScripts/Scripts/DomainReputation/DomainReputation_test.py:28]: done

File: Packs/CommonScripts/Scripts/DomainReputation/DomainReputation.py (hunk 3 above)
Comment (thefrieddan1) [reply Packs/CommonScripts/Scripts/DomainReputation/DomainReputation.py:17]: Agree

File: Packs/CommonScripts/ReleaseNotes/1_9_8.md (hunk 1 above)
Comment (thefrieddan1) [reply Packs/CommonScripts/ReleaseNotes/1_9_8.md:3]: done.

File: Packs/CommonScripts/Scripts/DomainReputation/DomainReputation.py (hunk 2 above)
Comment (thefrieddan1) [reply Packs/CommonScripts/Scripts/DomainReputation/DomainReputation.py:16]: done.

File: Packs/CommonScripts/Scripts/DomainReputation/DomainReputation_test.py [hunk 7]
```diff
@@ -19,3 +19,23 @@ def test_domain_reputation(mocker):
     domain_reputation()
     assert execute_mock.call_count == 1
     assert 'returned an error' in results_mock.call_args[0][0][0]['Contents']
+    from DomainReputation import domain_reputation
```
[the remark this answers is not in the document]
Comment (thefrieddan1) [reply Packs/CommonScripts/Scripts/DomainReputation/DomainReputation_test.py]: done.

**Review by ShahafBenYakir — APPROVED** [review]

**Review by ShahafBenYakir — APPROVED** [review]

**Pull request merged**